### ISSUE TITLE
E2B2 doesn't seem to work ?

### ISSUE THREAD

**Issue by tchapi** [tulir/mautrix-signal#43]
Hi,

I have followed https://github.com/spantaleev/matrix-docker-ansible-deploy and deployed the Signal and Telegram bridges.
Everything works as expected, except E2BE.

I have created an issue on the repo https://github.com/spantaleev/matrix-docker-ansible-deploy/issues/806 but I guess it's more relevant for the issue to be here.

Here's the detail of the error I have :

![105491855-52a13800-5cb7-11eb-86ad-82925c464eef](https://user-images.githubusercontent.com/1944007/105510311-c7cc3780-5cce-11eb-9004-6740c8a2f91f.png)

In short, when I set the `allow` and `default` booleans, I get the above error. 

```
  bridge:
    encryption:
      allow: true
      default: true
```

The playbooks use `dock.mau.dev/tulir/mautrix-signal:latest` for the image — could it be that this image does not have the optional e2be plugin installed ?

I don't really know how to look to debug this

Thanks a lot for your time !
Best

**Comment by tchapi** [comment]
Hmm while looking at https://github.com/tulir/mautrix-signal/blob/master/Dockerfile#L37 I guess the optional requirements is correctly installed on the Docker image ... so it must be something else

**Comment by pimeys** [comment]
I'm having exactly the same error, following the same steps as you did. I'm now running without E2BE, but it would be nicer for the future to have it enabled.

**Comment by pimeys** [comment]
Also, when you

```bash
> journalctl -fu matrix-mautrix-signal
```

You'll see errors such as

```
Jan 21 18:49:53 matrix matrix-mautrix-signal[5870]: [2021-01-21 17:49:53,532] [DEBUG@mau.bridge.e2ee] Didn't find session v1hSzd195KYYrvZXHKCFOrjQe4/fT/md0NJoSxCZuLQ, waiting 5 seconds for session to arrive
```

I've tried to uninstall the signal bridge, deleting the database and user, then reinstalling with encryption on. And, it should be needed to relink your signal connection after doing so, but even with doing all of this I still end up getting errors like above, and no messages moving between synapse and signal.

**Comment by tchapi** [comment]
Yes we're on the same page.

@tulir if you have any idea to solve this, my instance is pretty new so I can test and break things !

**Comment by pimeys** [comment]
I'm also ready to break my signal bridge if needed. The need for this feature and the joy of having my own matrix instance goes above the pain of getting things to work!

**Comment by DELUUXE** [comment]
I seem to be having the same issue, both with this bridge and the mautrix-whatsapp bridge. incomming messages seem to be encrypted fine (this includes synced messages) on the whatsapp bridge (i haven't tried this on the signal bridge yet) but messages i try to send from my matrix client return in the same "Your message was not bridged: the bridge hasn't received the decryption keys."
_i am also using the same ansible-playbook_

**Comment by tulir** [comment]
The ansible playbook seems to generate bad registration files that don't work with e2be (because synapse is also broken). See "Registration file workaround" in https://github.com/tulir/mautrix-telegram/wiki/End‐to‐bridge-encryption

**Comment by tchapi** [comment]
Hi @tulir

Your registration file workaround is indicated for mautrix-telegram < v0.9.0 but the ansible playbook uses the latest version, with Synapse > 1.21. `login_shared_secret` is also correctly set, so I'm not sure I understand what could be wrong in the playbook ?

Thanks for your help on this

**Comment by tulir** [comment]
The playbook brings its own registration file instead of having the bridge generate one, that file has not applied the workaround mentioned in the wiki. Even the latest versions of Synapse are broken (matrix-org/synapse#5758), so the workaround is still required.

**Comment by wrenix** [comment]
Hey, i got the same problem -> i do not know that is excatly broken?
What i have to change on my registration.yaml to get it working again?

This Weeky does not exists anymore, has somebody a backup?:
> See "Registration file workaround" in [https://github.com/tulir/mautrix-telegram/wiki/End‐to‐bridge-encryption](https://github.com/tulir/mautrix-telegram/wiki/End%E2%80%90to%E2%80%90bridge-encryption)

**Comment by tulir** [comment]
Don't post on old closed issues. The place to debug setup issues is the Matrix room
